client/app/services/policy/index.js

```diff
@@ -0,0 +1,8 @@
+import DefaultPolicy from './DefaultPolicy';
+
+// eslint-disable-next-line import/no-mutable-exports
+export let policy = new DefaultPolicy();
+
+export function setPolicy(newPolicy) {
+  policy = newPolicy instanceof DefaultPolicy ? newPolicy : new DefaultPolicy();
```

Comment:
If @ranbena sees this, he will immediately tag this as #typescript-ftw... 🤓 

But that aside, I'm worried this will create some race conditions that will be hard to spot, in case some other object holds on to the DefaultPolicy and "ignores" updates. I know this is what we do with the others, but in the case of the others, the default value is null so it's easier to spot issues.

Why not extend the default object with the new one?

Reply:
Because just `extend`ing will lose `this` context for all methods - this is even more error-prone. I think I'll just throw an exception here.

Reply:
...or just remove this check - hope nobody will put any garbage there